### PR TITLE
Replaced usage of deprecated static field `DESCRIPTOR` and added `NoExternalUse`

### PR DESCRIPTION
**Pull request by StefanSpieker**
I checked another type of Spotbugs errors and stumbled upon `ST - ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD` which is found for the static `DESCRIPTOR` fields. This is a preparation for removing them
Replaced usage of deprecated static field `DESCRIPTOR`and put `@Restricted(NoExternalUse.class)` on places where it was missing. So this is a preparation for deleting them in the future.

### Proposed changelog entries

* Internal: Mark static `DESCRIPTOR` fields as `NoExternalUse` as a preparation of removing them

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/node_monitors/TemporarySpaceMonitor.java
Comment (timja) [inline core/src/main/java/hudson/node_monitors/TemporarySpaceMonitor.java:57]: can't we use? `ExtensionList.lookupSingleton(DiskSpace.class)`

File: core/src/main/java/hudson/node_monitors/SwapSpaceMonitor.java
Comment (basil) [inline core/src/main/java/hudson/node_monitors/SwapSpaceMonitor.java:91]: This is currently used in [Alibaba Cloud ECS](https://github.com/jenkinsci/alibabacloud-ecs-plugin/blob/7b287c7752990c9caae62d2f9909b65f2092095e/src/main/java/com/alibabacloud/jenkins/ecs/monitor/FreeMemMonitor.java#L66). That being said, I think it's OK to mark it as restricted as long as we also file a corresponding PR to update the usage in Alibaba Cloud ECS.

**Review by basil — COMMENTED** [review]
`hudson.tasks.Maven#DESCRIPTOR` is another instance of a `DESCRIPTOR` field that should be marked with `@Restricted(NoExternalUse.class)`. Although [the comment](https://github.com/jenkinsci/jenkins/blob/87b2bc68ad105fd3d63cfac65c1bacdab64df5df/core/src/main/java/hudson/tasks/Maven.java#L421) reads "TODO: fix sonar plugin that depends on this. That's the only plugin that depends on this field.", that problem appears to have been fixed [since 2009](https://github.com/SonarSource/sonar-scanner-jenkins/commit/73bf76a).

**Comment by basil** [comment]
I have added the `stalled` label pending the review feedback being addressed. If the review feedback remains unaddressed after approximately one month, this PR will transition from the `stalled` label to the `proposed-for-close` label and closed approximately one week after that.

File: core/src/main/java/hudson/node_monitors/TemporarySpaceMonitor.java
Comment (basil) [reply core/src/main/java/hudson/node_monitors/TemporarySpaceMonitor.java]: Answer: No.

```
java.lang.IllegalStateException: Expected 1 instance of hudson.node_monitors.DiskSpaceMonitorDescriptor$DiskSpace but got 0
        at hudson.ExtensionList.lookupSingleton(ExtensionList.java:454)
        at hudson.node_monitors.TemporarySpaceMonitor.getFreeSpace(TemporarySpaceMonitor.java:65)
        at hudson.node_monitors.TemporarySpaceMonitor.getColumnCaption(TemporarySpaceMonitor.java:73)
        at hudson.model.ComputerSet.doUpdateNow(ComputerSet.java:234)
        at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
        at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:397)
        at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:409)
        at org.kohsuke.stapler.interceptor.RequirePOST$Processor.invoke(RequirePOST.java:78)
        at org.kohsuke.stapler.PreInvokeInterceptedFunction.invoke(PreInvokeInterceptedFunction.java:26)
        at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:207)
        at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:140)
        at org.kohsuke.stapler.MetaClass$11.doDispatch(MetaClass.java:558)
        at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:59)
        at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:766)
```

Reverting the commit for this non-suggestion.

**Comment by basil** [comment]
I just finished this PR myself.

**Review by basil — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by StefanSpieker** [comment]
@basil Thanks for finishing this PR. I was in the final phase of building my house and was really occupied for the last few months, but this should get better now. I'm really sorry for the inconveniences.

**Pull request merged**
